instrumentation/go.mongodb.org/mongo-driver/v2/mongo/otelmongo/mongo_test.go

```diff
@@ -24,6 +24,75 @@ import (
 
 type validator func(sdktrace.ReadOnlySpan) bool
 
+func TestWithSpanNameFormatter(t *testing.T) {
+	tt := []struct {
+		title             string
+		operation         func(context.Context, *mongo.Database) (any, error)
+		expectedSpanName  string
+		mockResponses     []bson.D
+		SpanNameFormatter SpanNameFormatterFunc
+	}{
+		{
+			title: "insert",
+			operation: func(ctx context.Context, db *mongo.Database) (any, error) {
+				return db.Collection("test-collection").InsertOne(ctx, bson.D{{Key: "test-item", Value: "test-value"}})
+			},
+			expectedSpanName:  "test-collection.insert",
+			SpanNameFormatter: nil,
+			mockResponses:     []bson.D{{{Key: "ok", Value: 1}}},
+		},
+		{
+			title: "delete",
```

Comment:
[nit] Could we title this w/ the context of using a non-default formatter?

```suggestion
			title: "delete with custom SpanNameFormatter",
```